typescript/integrations.d.ts

```diff
@@ -302,6 +303,22 @@ export type IntegrationCivicRec<S = SavedSecret> = IntegrationBase & {
   }
 }
 
+export type IntegrationOAuth2Provider<S = SavedSecret> = {
+  id: string
+  label: string
+  clientId: string
+  clientSecret: ConstrainedSecret<S>
+  attributeRequestMethod: 'GET' | 'POST'
+  issuerURL: string
+}
+
+export type IntegrationOAuth2<S = SavedSecret> = IntegrationBase & {
+  type: 'OAUTH_2'
+  configuration: {
+    providers: IntegrationOAuth2Provider<S>[]
+  }
+}
+
```

Comment:
I didn't expect we would be storing this information in integrations but I guess we could 🤔

Reply:
I think it makes sense.